File: malevich-server/src/main/java/io/malevich/server/transfer/ArtworkStockDto.java
package io.malevich.server.transfer;

import com.fasterxml.jackson.annotation.JsonView;
import com.yinyang.core.server.core.dto.View;
import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
public class ArtworkStockDto {

    private Long id;

    private ArtworkDto artwork;

    private GalleryDto gallery;

    @JsonView({View.Admin.class, View.Gallery.class, View.Trader.class})
    private Double instantPrice;

    @JsonView({View.Admin.class, View.Gallery.class, View.Trader.class})
    private Double bestBid;

    @JsonView({View.Admin.class, View.Gallery.class, View.Trader.class})
    private Double lastPrice;

}

File: malevich-web/src/app/_transfer/artworkStockDto.ts
import {GalleryDto} from "./galleryDto";
import {ArtworkDto} from "./artworkDto";

export class ArtworkStockDto {
  id: number;
  artwork: ArtworkDto;
  gallery: GalleryDto;
  instantPrice: number;
  bestBid: number;
  lastPrice: number;
}
